src/Webhook/Mappers/ShipmentItemsMapper.php

```diff
@@ -53,17 +51,31 @@ public function handle( array $item ) {
 
             $this->loadEntity($shipmentItemId, $trackMageId);
 
-            if(($res = $this->canHandle()) < 0)
-                return $res;
+            if($this->canHandle())
+                return false;
 
             $data = $this->prepareData();
 
             $this->entity = $this->repo->update($data, ['id' => $shipmentItemId]);
 
-            return $this->entity;
         }catch (\Throwable $e){
             throw new EndpointException('An error happened during update shipment from TrackMage: '.$e->getMessage(), $e->getCode(), $e);
         }
     }
 
+    protected function prepareData() {
+        $data = parent::prepareData();
+
+        if(isset($data["orderItem"])){
+            $trackmageOrderItemId = str_replace('/order_items/','', $data["orderItem"]);
+            global $wpdb;
+            $row = $wpdb->get_row("SELECT * FROM ".$wpdb->prefix . 'woocommerce_order_itemmeta'." WHERE meta_key = '_trackmage_order_item_id' AND meta_value = '".$trackmageOrderItemId."'", ARRAY_A);
+            if(is_array($row) && isset($row['order_item_id']))
+                $data["orderItem"] = $row['order_item_id'];
+            else
+                throw new EndpointException('Order item was not found.',400);
+        }
+        return $data;
```

Comment:
can you please extract it to a separate function:
`$data['order_item_id'] = $this->getOrderItemIdByTrackMageId($trackmageOrderItemId);`